de1plus/bluetooth.tcl

```diff
@@ -946,6 +948,45 @@ proc decentscale_tare {} {
 
 }
 
+#### Smartchef Scale
+proc smartchef_enable_weight_notifications {} {
+	if {$::de1(scale_device_handle) == 0 || $::settings(scale_type) != "smartchef"} {
+		return
+	}
+
+	if {[ifexists ::sinstance($::de1(suuid_smartchef))] == ""} {
+		::bt::msg -DEBUG "decent scale not connected, cannot enable weight notifications"
+		return
+	}
+
+	userdata_append "SCALE: enable smartchef scale weight notifications" [list ble enable $::de1(scale_device_handle) $::de1(suuid_smartchef) $::sinstance($::de1(suuid_smartchef)) $::de1(cuuid_smartchef_status) $::cinstance($::de1(cuuid_smartchef_status))] 0
+}
+
+proc smartchef_tare {} {
+	#software based taring is not supported by the scale
+	if {$::de1(scale_device_handle) == 0 || $::settings(scale_type) != "smartchef"} {
+		return
+	}
+
+	if {[ifexists ::sinstance($::de1(suuid_smartchef))] == ""} {
+		error "Smartchef Scale not connected, cannot send tare cmd"
```

Comment:
Will this show up as a user notification? If so if might be useful to change this to be instructional: Warning: Auto-Tare not available, please hit Tare on the scale"

Reply:
Well, I wasn't sure about that. Doesn't that get annoying to see every time? An alternative would be to save the weight at taring and subtract it from the current weight. This way the app would have an "soft" tare. I implemented this behavior in  despresso but didn't dare to do it here. I wanted to make it as simple as possible.